pegasus/sites.v3/code.org/public/private/k5survey/index.haml

```diff
@@ -22,119 +19,389 @@
+      careers_knowledge_total: careers_knowledge_total,
+      comments: comments
+    }
+  end
 
-        # get data on students of the teacher
-        teacher_user_id = teacher[:id]
-        next unless teacher_user_id
+  if facilitator_id
+    workshops = DB[:forms].where(kind:'ProfessionalDevelopmentWorkshop', user_id:facilitator_id)
 
-        students = DASHBOARD_DB[:followers].
-          where(user_id: teacher_user_id).
-          join(:users, id: :student_user_id).
-          select(:users__id___id, :users__created_at___created_at)
+    query = "SELECT * FROM forms WHERE kind='K5ProfessionalDevelopmentSurvey' AND "
+    workshops.each do |workshop|
+      query += "data LIKE '%\"workshop_id_i\":\""
+      query += workshop[:id].to_s + "\"%\' OR "
+    end
+    query = query[0, query.length - 3]
 
-        if students.count > 0
-          lifetime = students.map{|s| (Time.now - s[:created_at]) / (60 * 60 * 24)}.reduce(:+) / students.count.to_f
+    individual_surveys = DB.fetch(query)
 
-          levels = students.map do |s|
-            DASHBOARD_DB[:user_levels].
-            where(user_id: s[:id]).
-            and("best_result >= #{ActivityConstants::MINIMUM_PASS_RESULT}").
-            count
-          end.reduce(:+) / students.count.to_f
-        else
-          lifetime = 0
-          levels = 0
-        end
+    individual_survey_totals = total_survey_data(individual_surveys)
+    all_surveys_totals = total_survey_data(all_surveys)
 
-        {
-          teacher_name: teacher[:name],
-          teacher_email: teacher[:email],
-          affiliate_name: affiliate[:name],
-          affiliate_email: affiliate[:user_id],
-          students_count: students.count,
-          students_average_lifetime_days: lifetime.round,
-          students_average_levels_completed: levels.round(2)
-        }
-      end.compact
-    end.compact.flatten
+    total_attendees = 0
+    workshops.each do |workshop|
+      data = JSON.parse(workshop[:data]).merge(JSON.parse(workshop[:processed_data]))
+      if data['total_attendee_count_i']
+        total_attendees += data['total_attendee_count_i']
+      end
+    end
   end
-%link(rel='stylesheet' type='text/css' href='/css/learn_tabs.css')
+
+:javascript
+  $(document).ready(function() {
+    $('select').bind("change", function () {
+      window.location = $("select option:selected").first().val();
+    });
+  });
+
 %h1
-  K5 Survey Statistics
-#learn-tabs{style: "padding-top:20px; padding-left:20px; padding-right:20px; background-color:white; background-color: rgb(219, 219, 219); overflow: hidden; margin-bottom: 10px;"}
-  %ul.nav.nav-tabs{style: "float: left;"}
-    %li.active.tab
-      %a{href:"/"} Results
-    %li
-      %a{href:"k5survey/raw2"} Raw2
+  K-5 Facilitator Profile
+%select
+  %option{value:"/private/k5survey"}
+    Select a Facilitator
+  - all_facilitators.each do |facilitator|
+    %option{value:"#{'/private/k5survey?user_id=' + facilitator[:user_id].to_s}"}
```

Comment:
"#{}" is redundant, you should be able to just do 
`value: '/private/k5survey?user_id=' + facilitator[:user_id].to_s`